### ISSUE TITLE
[Bug]: Ku-ring-gai Council source error?

### ISSUE THREAD

**Issue by jonwaland** [mampfes/hacs_waste_collection_schedule#2846]
### I Have A Problem With:

A specific source

### What's Your Problem

```
2024-10-15 11:34:55.669 ERROR (MainThread) [homeassistant] Error doing job: Future exception was never retrieved (None)
Traceback (most recent call last):
  File "/usr/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/homeassistant/.homeassistant/custom_components/waste_collection_schedule/waste_collection_api.py", line 102, in _fetch
    shell.fetch()
  File "/home/homeassistant/.homeassistant/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 155, in fetch
    e.set_type(e.type.strip())
               ^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'strip'
```

during testing with test_sources.py it looks like the "General Waste" and "Green waste" returns from council have changed? 

### Source (if relevant)

kuringgai_nsw_gov_au

### Logs

```Shell
2024-10-15 11:34:55.669 ERROR (MainThread) [homeassistant] Error doing job: Future exception was never retrieved (None)
Traceback (most recent call last):
  File "/usr/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/homeassistant/.homeassistant/custom_components/waste_collection_schedule/waste_collection_api.py", line 102, in _fetch
    shell.fetch()
  File "/home/homeassistant/.homeassistant/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 155, in fetch
    e.set_type(e.type.strip())
               ^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'strip'

(homeassistant) homeassistant@jonlabnuc:~/.homeassistant/custom_components/waste_collection_schedule/waste_collection_schedule/test$ python3 ./test_sources.py -s kuringgai_nsw_gov_au -i -l
Testing source kuringgai_nsw_gov_au ...
  found 3 entries for randomHouse
    2024-10-16 : None [None]
    2024-10-23 : Recycling [mdi:recycle]
    2024-10-16 : None [None]
  found 3 entries for randomAppartment
    2024-10-18 : None [None]
    2024-10-18 : Recycling [mdi:recycle]
    2024-10-18 : None [None]
  found 3 entries for randomMultiunit
    2024-10-17 : None [None]
    2024-10-17 : Recycling [mdi:recycle]
    2024-10-17 : None [None]
```

### Relevant Configuration

```YAML
waste_collection_schedule:
  sources:
    - name: kuringgai_nsw_gov_au
      args:
        post_code: "2073"
        suburb: PYMBLE
        street_name: 'SomeStreet'
        street_number: "9999"
      customize:
        - type: 'General Waste'
          icon: mdi:trash-can
        - type: Recycling
          icon: mdi:recycle
        - type: Green Waste
          icon: mdi:leaf
```

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [X] Checked that the website of your service provider is still working
- [X] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [X] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by jonwaland** [comment]
Ahh - a subtle change - I did look at the data returned by the geo api but couldn't work out what had changed.

**Comment by 5ila5** [comment]
fixed by #2851

This will be part of the next release or is available now as master (git) version 

install master/git: Developer tools -> actions -> select `update: Install update` -> click `Coose entity` and select WCS -> select `version` and type `master` in the input field -> run the action -> restart HA


<details><summary>Older versions of HACS:</summary>
<p>
install master with HACS version < 2.0: (3-dot menu -> redownload -> change version to master -> fully restart HA (this will increase update cycle by a lot)
</p>
</details>
